packages/flutter/lib/src/painting/image_cache.dart

```diff
@@ -28,6 +28,57 @@ const int _kDefaultSizeBytes = 100 << 20; // 100 MiB
 ///
 /// A shared instance of this cache is retained by [PaintingBinding] and can be
 /// obtained via the [imageCache] top-level property in the [painting] library.
+///
+/// {@tool sample}
+///
+/// This sample shows how to supply your own caching logic and replace the
+/// global [imageCache] varible.
+///
+/// ```dart
+/// import 'package:flutter/foundation.dart' as foundation;
+/// import 'package:flutter/gestures.dart' as gestures;
+/// import 'package:flutter/rendering.dart' as rendering;
+/// import 'package:flutter/scheduler.dart' as scheduler;
+/// import 'package:flutter/services.dart' as services;
+/// import 'package:flutter/widgets.dart' as widgets;
+///
+/// class MyImageCache extends widgets.ImageCache {
+///   @override
+///   void clear() {
+///     print("Clearing cache!");
+///     super.clear();
+///   }
+/// }
+///
+/// class MyWidgetsBinding extends foundation.BindingBase
+///     with
+///         gestures.GestureBinding,
+///         services.ServicesBinding,
+///         scheduler.SchedulerBinding,
+///         rendering.PaintingBinding,
+///         rendering.SemanticsBinding,
+///         rendering.RendererBinding,
+///         widgets.WidgetsBinding {
+///
+///   @override
+///   widgets.ImageCache createImageCache() => MyImageCache();
+/// }
+///
+/// void main() {
+///   // Magic happens in the constructor to set global variables.
```

Comment:
nit: I wouldn't call this "magic". Just say what it does.

Reply:
Done.  IMO there should be some further qualifier that calls out this isn't what you'd normally expect from a constructor, since typically the only side effects they have is on the object that is being constructed.